libraries/AC_PID/AC_PID_2D.cpp

```diff
@@ -133,16 +133,17 @@ void AC_PID_2D::update_i(const Vector2f &limit)
     _pid_info_y.limit = false;
 
     Vector2f delta_integrator = (_error * _ki) * _dt;
+    float integrator_length = _integrator.length();
+    _integrator += delta_integrator;
     if (!is_zero(limit.length_squared())) {
```

Comment:
```suggestion
    if (!is_zero(integrator_length)) {
```
small efficiency improvement.

Reply:
Agreed but I will move that down to remove the sqrt unless we need it.

Reply:
This is actually the correct change. Sorry.